File: app/src/main/java/ru/systempla/weatherapp/mvp/view/WeatherDataView.java
package ru.systempla.weatherapp.mvp.view;

import moxy.MvpView;
import moxy.viewstate.strategy.AddToEndSingleStrategy;
import moxy.viewstate.strategy.StateStrategyType;
import moxy.viewstate.strategy.OneExecutionStateStrategy;

@StateStrategyType(value = AddToEndSingleStrategy.class)
public interface WeatherDataView extends MvpView {

    void setCityName(String name);
    void setWeatherDescription(String description);
    void setUVIndex (float uvIndex);
    void setPressure(float pressure);
    void setHumidity(float humidity);
    void setWindSpeed(double speed);
    void setWeatherIcon(int actualId, long sunrise, long sunset);
    void setCurrentTemperature(float temp);
    void showLoading();
    void hideLoading();

    @StateStrategyType(value = OneExecutionStateStrategy.class)
    void showMessage(String text);
}

File: app/src/main/java/ru/systempla/weatherapp/mvp/view/WeatherDataView.kt
package ru.systempla.weatherapp.mvp.view

import moxy.MvpView
import moxy.viewstate.strategy.AddToEndSingleStrategy
import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType

@StateStrategyType(value = AddToEndSingleStrategy::class)
interface WeatherDataView : MvpView {
    fun setCityName(name: String)
    fun setWeatherDescription(description: String)
    fun setUVIndex(uvIndex: Float)
    fun setPressure(pressure: Float)
    fun setHumidity(humidity: Float)
    fun setWindSpeed(speed: Double)
    fun setWeatherIcon(actualId: Int, sunrise: Long, sunset: Long)
    fun setCurrentTemperature(temp: Float)
    fun showLoading()
    fun hideLoading()
    fun checkForGPSUpdate()

    @StateStrategyType(value = OneExecutionStateStrategy::class)
    fun showMessage(text: String?)
}